crates/subspace-fraud-proof/src/invalid_state_transition_proof.rs

```diff
@@ -449,7 +449,26 @@ where
             heap_pages: None,
         };
 
-        let call_data = b"TODO: fetch call_data from the original primary block";
+        let call_data = match execution_phase {
+            ExecutionPhase::InitializeBlock { domain_parent_hash } => {
+                let parent_hash =
+                    <Block as BlockT>::Hash::decode(&mut domain_parent_hash.encode().as_slice())?;
+                let parent_number =
+                    <NumberFor<Block>>::decode(&mut parent_number.encode().as_slice())?;
+                let new_header = <Block as BlockT>::Header::new(
+                    parent_number,
+                    Default::default(),
+                    Default::default(),
+                    parent_hash,
+                    Default::default(),
```

Comment:
Thanks for the comment, yes, I did forget to add a TODO, rebased and pushed a commit to add it. The code changes to resolve it won't be significant, but we need to add a new test to cover the regular fraud proof creation and verification route. I expect you to take that, but I could also work on it once I process the other priorities.

Reply:
> but we need to add a new test to cover the regular fraud proof creation and verification route. I expect you to take that, but I could also work on it once I process the other priorities.

Sure, it is on my plan for this week.

Reply:
>are we not handling the proper digests in this PR ?

Nope. We found the system domain digests issue on the proof creation as described [here](https://github.com/subspace/subspace/pull/1344/commits/b5dca024e92e27eb98377ab0eb0ac3a060408f5c#diff-77459f83c5cf59cea043e7ca65b282ab5ff94c1ac8b463dda1588cdef8ba7d80R675) before and agreed @NingLin-P will take care of it with a test added. Since the creation and verification are basically in one single test flow, I'll leave it as is to Ning.

Reply:
The digest issue of system domain has been addressed in the latest commit, I had to fix it right now to pass the test.

Reply:
The fraud proof test is also ready here https://github.com/subspace/subspace/tree/fraud-proof-test, I will open a PR for it after this PR is merged.